### PR TITLE
chore(ordered-list): add ordered list controls

### LINKED ISSUE #12146: Ensure order list has working controls

### PR DESCRIPTION
**Pull request by aledavila**
Closes #12146

Adds controls to Ordered List and fixed story issue

#### Changelog

**New**

- controls

**Removed**

- removed stories that can be accessed with controls

#### Testing / Reviewing

Make sure 'Story' tab shows code and controls work as intended

### CODE REVIEW

File: packages/react/src/components/OrderedList/OrderedList-story.js [hunk 1]
```diff
@@ -1,92 +0,0 @@
...
-    <ListItem>Ordered List level 1</ListItem>
-    <ListItem>Ordered List level 1</ListItem>
-    <ListItem>Ordered List level 1</ListItem>
-    <ListItem>Ordered List level 1</ListItem>
-    <ListItem>Ordered List level 1</ListItem>
-    <ListItem>Ordered List level 1</ListItem>
-  </OrderedList>
-);
-
-export const Nested = () => {
```
Comment (tay1orjones) [inline packages/react/src/components/OrderedList/OrderedList-story.js:41]: What's the reason for removing the Nested and NativeListStyles stories?

Also percy is failing - for any story name changes the `e2e/components/OrderedList/OrderedList-test.e2e.js` file will need updated.

Comment (aledavila) [reply packages/react/src/components/OrderedList/OrderedList-story.js:41]: @tay1orjones I can add them back I thought it was redundant due to the controls but no problem.

**Comment by aledavila** [comment]
@tay1orjones updated ! ready to review again

File: packages/react/src/components/OrderedList/OrderedList-story.js (hunk 1 above)
Comment (tay1orjones) [reply packages/react/src/components/OrderedList/OrderedList-story.js:41]: Cool, yeah I think it's helpful from a VRT perspective to have those in there to catch any regressions

**Review by tay1orjones — APPROVED** [review]

**Review by abbeyhrt — APPROVED** [review]

**Pull request merged**